File: KthLargestMinHeap.java
// Time Complexity : O(nlog(k))
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this :    No
package heap1;

import java.util.*;

public class KthLargestMinHeap {

    public int findKthLargest(int[] nums, int k) {
        //pq is ascending by default, for desc we can write (a,b) -> b -a
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> a - b);
        for (int num : nums) {
            pq.add(num);
            if (pq.size() > k) {
                pq.poll();
            }
        }
        return pq.peek();
    }
}

File: kthlargestminheap.py
# // Time Complexity :O(nlogk)
# // Space Complexity :O(k),elements in the heap
# // Did this code successfully run on Leetcode :yes
# // Any problem you faced while coding this :no


# // Your code here along with comments explaining your approach

from heapq import heappush as insert
from heapq import heappop as remove
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        minheap=[]
        for i in range(len(nums)):
            insert(minheap,nums[i])
            # print(minheap)
            if len(minheap)>k:
                remove(minheap)
        return minheap[0]
            
            
        